internal/api/core/user_test.go

```diff
@@ -0,0 +1,232 @@
+package core
+
+import (
+	"reflect"
+	"slices"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/model"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestNewUserInfos(test *testing.T) {
```

Comment:
Consider adding some more cases. Ex. testing if the user isn't enrolled, testing server admin/owner -> course owner, etc.